src/Player/Camera/Camera.gd

```diff
@@ -1,7 +1,86 @@
-extends Camera
+extends Spatial
 
 var _target: Spatial
+export var camera_rotation_speed: = Vector2(2.5, 2.5)
+export var yinvert_multiplier: = -1
+export var camera_backwards_deadzone := 0.3
+var _camera_rotating: = false
+
+onready var initial_camera_pos = $Camera.get_translation()
+onready var initial_anchor_pos = get_translation()
+onready var rotation_delay: Timer = $RotateDelay
+onready var occlusion_ray: RayCast = $OcclusionRay
 
 func _ready() -> void:
 	_target = owner
 	set_as_toplevel(true)
+	occlusion_ray.set_cast_to(initial_camera_pos)
+
+
+func physics_process(delta: float, move_direction: Vector3) -> void:
+	#TODO: Remove move direction visual aid - or turn it into a gizmo?
+	$MoveDirectionRef.set_translation(move_direction)
```

Comment:
All node dependencies should be `onready var ...` at the top of the variable definitions